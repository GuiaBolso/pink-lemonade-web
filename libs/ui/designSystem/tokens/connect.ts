/**
 * TODO: configure eslint to enable internal import
 */
import utils from '@pink-lemonade-web/utils'; // eslint-disable-line

import tokens from './guiaBolso';

const { pxToRem } = utils.scales;

const baseFontSize = 16;

export default {
  ...tokens,

  type: {
    ...tokens.type,

    baseFontSize,

    sizes: {
      fontSize12: pxToRem(12, baseFontSize),
      fontSize14: pxToRem(14, baseFontSize),
      fontSize16: pxToRem(16, baseFontSize),
      fontSize18: pxToRem(18, baseFontSize),
      fontSize20: pxToRem(20, baseFontSize),
      fontSize24: pxToRem(24, baseFontSize),
      fontSize32: pxToRem(32, baseFontSize),
      fontSize36: pxToRem(36, baseFontSize),
      fontSize42: pxToRem(42, baseFontSize),
      fontSize48: pxToRem(48, baseFontSize),
      fontSize52: pxToRem(52, baseFontSize),
      fontSize64: pxToRem(64, baseFontSize),
      fontSize68: pxToRem(68, baseFontSize),
      fontSize80: pxToRem(80, baseFontSize),
    },

    fontWeight: {
      bold: 700,
      semibold: 600,
      regular: 400,
      light: 300,
    },
  },

  colors: {
    ...tokens.colors,

    brand: {
      ...tokens.colors.brand,

      secondary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },

    neutral: {
      blank: '#ffffff',
      lightest: '#f9f9fa',
      lighter: '#f3f4f4',
      light: '#eeeeef',
      brand: '#e2e3e4',
      dark: '#888889',
      darker: '#5a5b5b',
      darkest: '#2d2d2e',
    },

    font: {
      brand: {
        primary: '#250048',
        secondary: '#00c1eb',
        neutral: '#5a5b5b',
      },
    },
  },
};
