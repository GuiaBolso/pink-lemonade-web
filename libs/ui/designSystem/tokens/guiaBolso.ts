/**
 * TODO: configure eslint to enable internal import
 */
import utils from '@pink-lemonade-web/utils'; // eslint-disable-line

const { pxToRem } = utils.scales;

const baseFontSize = 16;

export type TokensGuiaBolsoProps = {
  name?: string;
  type?: {
    baseFontSize?: string | number;
    sizes?: {
      xxxl?: string;
      xxl?: string;
      xl?: string;
      lg?: string;
      md?: string;
      sm?: string;
      xs?: string;
      xxs?: string;
      xxxs?: string;
    };
    lineHeight?: {
      rule1?: string;
      rule2?: string;
      rule3?: string;
      rule4?: string;
    };
    fontFamily?: {
      brand?: string;
      neutral?: string;
    };
    fontWeight?: {
      bold?: string | number;
      semibold?: string | number;
      regular?: string | number;
    };
  };
  colors?: {
    brand?: {
      primary?: {
        lightest?: string;
        lighter?: string;
        light?: string;
        default?: string;
        dark?: string;
        darker?: string;
      };
      secondary?: {
        lightest?: string;
        lighter?: string;
        light?: string;
        default?: string;
        dark?: string;
        darker?: string;
      };
      tertiary?: {
        lightest?: string;
        lighter?: string;
        light?: string;
        default?: string;
        dark?: string;
        darker?: string;
      };
    };

    neutral?: {
      blank?: string;
      lightest?: string;
      light?: string;
      regular?: string;
      medium?: string;
      dark?: string;
      darkest?: string;
    };

    background?: {
      neutral?: {
        lightest?: string;
        blank?: string;
      };
      brand?: {
        primary?: string;
        secondary?: string;
        tertiary?: string;
      };
    };

    feedback?: {
      error?: {
        default?: string;
        lightest?: string;
      };
      success?: {
        default?: string;
        lightest?: string;
      };
      alert?: {
        default?: string;
        lightest?: string;
      };
      notification?: {
        default?: string;
        lightest?: string;
      };
    };

    font?: {
      brand?: {
        primary?: string;
        secondary?: string;
        neutral?: string;
      };
    };
  };
  breakpoints?: any;
  zIndex?: any;
  spacing?: any;
};

const tokens: TokensGuiaBolsoProps = {
  name: 'guiaBolso',

  type: {
    baseFontSize,

    sizes: {
      xxxl: pxToRem(64, baseFontSize),
      xxl: pxToRem(48, baseFontSize),
      xl: pxToRem(40, baseFontSize),
      lg: pxToRem(32, baseFontSize),
      md: pxToRem(24, baseFontSize),
      sm: pxToRem(20, baseFontSize),
      xs: pxToRem(18, baseFontSize),
      xxs: pxToRem(14, baseFontSize),
      xxxs: pxToRem(12, baseFontSize),
    },

    lineHeight: {
      rule1: 'calc(var(--fontSize) + 4px)',
      rule2: 'calc(var(--fontSize) + 8px)',
      rule3: 'calc(var(--fontSize) + 12px)',
      rule4: 'calc(var(--fontSize) + 14px)',
    },

    fontFamily: {
      brand: 'Axiforma',
      neutral: 'Open Sans',
    },

    fontWeight: {
      bold: 'bold',
      semibold: 'semibold',
      regular: 'regular',
    },
  },

  colors: {
    brand: {
      primary: {
        lightest: '#f4f2f6',
        lighter: '#715988',
        light: '#462663',
        default: '#250048',
        dark: '#21023f',
        darker: '#1d0533',
      },
      secondary: {
        lightest: '#fff5f8',
        lighter: '#fa688e',
        light: '#f94876',
        default: '#f9386a',
        dark: '#d6325c',
        darker: '#a62949',
      },
      tertiary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },

    neutral: {
      blank: '#ffffff',
      lightest: '#f2f2f2',
      light: '#d9d9d9',
      regular: '#aaaaaa',
      medium: '#737373',
      dark: '#404040',
      darkest: '#0d0d0d',
    },

    background: {
      neutral: {
        lightest: '#fafafa',
        blank: '#ffffff',
      },
      brand: {
        primary: '#250048',
        secondary: '#f9386a',
        tertiary: '#00c1eb',
      },
    },

    feedback: {
      error: {
        default: '#e20b10',
        lightest: '#fef3f3',
      },
      success: {
        default: '#19bd54',
        lightest: '#f3fcf6',
      },
      alert: {
        default: '#fec01a',
        lightest: '#fffcf4',
      },
      notification: {
        default: '#005eeb',
        lightest: '#f2f7fe',
      },
    },

    font: {
      brand: {
        primary: '#f9386a',
        secondary: '#3f1b68',
        neutral: '#404040',
      },
    },
  },

  breakpoints: {
    key: '',
  },

  zIndex: {
    key: 10,
  },

  spacing: {
    scale: [],
  },
};

export default tokens;
