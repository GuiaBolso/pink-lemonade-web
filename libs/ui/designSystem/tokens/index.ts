/**
 * TODO: configure eslint to enable internal import
 */
import utils from '@pink-lemonade-web/utils'; // eslint-disable-line

const { pxToRem } = utils.scales;

const baseFontSize = 16;

export default {
  type: {
    baseFontSize,

    sizes: {
      xxxl: pxToRem(64, baseFontSize),
      xxl: pxToRem(48, baseFontSize),
      xl: pxToRem(40, baseFontSize),
      lg: pxToRem(32, baseFontSize),
      md: pxToRem(24, baseFontSize),
      sm: pxToRem(20, baseFontSize),
      xs: pxToRem(18, baseFontSize),
      xxs: pxToRem(14, baseFontSize),
      xxxs: pxToRem(12, baseFontSize),
    },

    lineHeight: {
      rule1: 'calc(var(--fontSize) + 4px)',
      rule2: 'calc(var(--fontSize) + 8px)',
      rule3: 'calc(var(--fontSize) + 12px)',
      rule4: 'calc(var(--fontSize) + 14px)',
    },

    fontFamily: {
      brand: 'Axiforma',
      neutral: 'Open Sans',
    },

    fontWeight: {
      bold: 'bold',
      semibold: 'semibold',
      regular: 'regular',
    },
  },

  colors: {
    brand: {
      primary: {
        lightest: '#f4f2f6',
        lighter: '#715988',
        light: '#462663',
        default: '#250048',
        dark: '#21023f',
        darker: '#1d0533',
      },
      secondary: {
        lightest: '#fff5f8',
        lighter: '#fa688e',
        light: '#f94876',
        default: '#f9386a',
        dark: '#d6325c',
        darker: '#a62949',
      },
      tertiary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },

    neutral: {
      blank: '#ffffff',
      lightest: '#f2f2f2',
      light: '#d9d9d9',
      regular: '#aaaaaa',
      medium: '#737373',
      dark: '#404040',
      darkest: '#0d0d0d',
    },

    background: {
      neutral: {
        lightest: '#fafafa',
        blank: '#ffffff',
      },
      brand: {
        primary: '#250048',
        secondary: '#f9386a',
        tertiary: '#00c1eb',
      },
    },

    feedback: {
      error: {
        default: '#e20b10',
        lightest: '#fef3f3',
      },
      success: {
        default: '#19bd54',
        lightest: '#f3fcf6',
      },
      alert: {
        default: '#fec01a',
        lightest: '#fffcf4',
      },
      notification: {
        default: '#005eeb',
        lightest: '#f2f7fe',
      },
    },

    font: {
      brand: {
        primary: '#f9386a',
        secondary: '#3f1b68',
        neutral: '#404040',
      },
    },
  },

  breakpoints: {
    key: '',
  },

  zIndex: {
    key: 10,
  },

  spacing: {
    scale: [],
  },
};
