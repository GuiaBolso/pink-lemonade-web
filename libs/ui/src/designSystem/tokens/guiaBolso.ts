import { pxToRem } from '@guiabolsobr/utils';
import { DeepPartial } from '../../typings/DeepPartial';

const baseFontSize = 16;

export type TokensGuiaBolsoProps = DeepPartial<typeof tokens>;

const tokens = {
  name: 'guiaBolso',

  type: {
    baseFontSize,

    sizes: {
      xxxxxl: pxToRem(80, baseFontSize),
      xxxxl: pxToRem(64, baseFontSize),
      xxxl: pxToRem(48, baseFontSize),
      xxl: pxToRem(40, baseFontSize),
      xl: pxToRem(36, baseFontSize),
      lg: pxToRem(32, baseFontSize),
      md: pxToRem(24, baseFontSize),
      sm: pxToRem(20, baseFontSize),
      xs: pxToRem(18, baseFontSize),
      xxs: pxToRem(16, baseFontSize),
      xxxs: pxToRem(14, baseFontSize),
      xxxxs: pxToRem(12, baseFontSize),
      xxxxxs: pxToRem(10, baseFontSize),
    },

    lineHeight: {
      rule1: 'calc(var(--fontSize) + 4px)',
      rule2: 'calc(var(--fontSize) + 8px)',
      rule3: 'calc(var(--fontSize) + 12px)',
      rule4: 'calc(var(--fontSize) + 14px)',
    },

    fontFamily: {
      brand: 'Montserrat',
      neutral: 'Open Sans',
    },

    fontWeight: {
      bold: 700,
      semibold: 600,
      regular: 400,
      light: 300,
    },
  },

  colors: {
    brand: {
      primary: {
        lightest: '#f4f2f6',
        lighter: '#715988',
        light: '#462663',
        default: '#250048',
        dark: '#21023f',
        darker: '#1d0533',
      },
      secondary: {
        lightest: '#fff5f8',
        lighter: '#fa688e',
        light: '#f94876',
        default: '#f9386a',
        dark: '#d6325c',
        darker: '#a62949',
      },
      tertiary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },

    neutral: {
      blank: '#ffffff',
      lightest: '#f2f2f2',
      light: '#d9d9d9',
      regular: '#aaaaaa',
      medium: '#737373',
      dark: '#404040',
      darkest: '#0d0d0d',
    },

    background: {
      neutral: {
        lightest: '#fafafa',
        blank: '#ffffff',
      },
      brand: {
        primary: '#250048',
        secondary: '#f9386a',
        tertiary: '#00c1eb',
      },
    },

    feedback: {
      error: {
        default: '#e20b10',
        lightest: '#fef3f3',
      },
      success: {
        default: '#19bd54',
        lightest: '#f3fcf6',
      },
      alert: {
        default: '#fec01a',
        lightest: '#fffcf4',
      },
      notification: {
        default: '#005eeb',
        lightest: '#f2f7fe',
      },
    },
  },
};

export default tokens;
