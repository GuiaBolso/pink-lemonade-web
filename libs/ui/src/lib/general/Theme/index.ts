import { customThemeProps as TCustomThemeProps } from './Theme';

export { default as Theme } from './Theme';
export type customThemeProps = TCustomThemeProps;
