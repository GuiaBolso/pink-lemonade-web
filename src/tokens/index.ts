import DesignSystem from 'design-system-utils';

const designTokens: any = {
  type: {
    // this should be set as a px value if you have `options.fontSizeUnit` set
    // to 'rem' or 'em' so that the lib can convert the values properly
    baseFontSize: '16px', // string,

    // used with `tokens.fs('size')` or `tokens.fontSize('size')`
    sizes: {
      xs: '8px',
      s: '16px',
      m: '24px',
      l: '40px',
      xl: '56px',
      xxl: '80px',
      expanded: '100%',
    },

    fontFamily: {
      system:
        '"Open Sans", OpenSans, -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Oxygen, Ubuntu, Cantarell, "Fira Sans", "Droid Sans"',
      sans: '"Helvetica Neue", Helvetica, Arial, sans-serif',
      serif: 'Georgia, "Times New Roman", Times, serif',
      mono: 'Menlo, Monaco, "Courier New", monospace',
    },

    lineHeight: {
      headings: 1.1,
    },

    fontWeight: {
      normal: 300, // Useful to set here if using anything other than `normal`
      bold: 'bold', // Useful to set here when bold webfonts come as 400 font-weight.
      headings: 'bold', // instead of browser default, bold
    },
  },

  // Color palette
  // Each object needs to have the same shape
  // Each color object needs a `base` value to be the default
  // Have as many color objects as you like
  colors: {
    // Used with `ds.color('colorName')`
    colorPalette: {
      neutral: {
        blank: '#ffffff', // <string>, // base is the default
        lightest: '#0000000c',
      },
      text: {
        base: '#000000bf',
        blank: '#ffffff',
        lightest: '#00000026',
        lighter: '#00000059',
        light: '#00000059',
        dark: '#000000f2',
      },
      feedback: {
        error: '#d50000',
        success: '#57b45b',
        warning: '#e8b44d',
      },
      accent: {
        primary: '#f9386a',
        secondary: '#604579',
      },
      finance: {
        income: '#00bd6e',
      },
    },

    // Used with `ds.brand('colorName)`
    brand: {
      primary: '#250048',
      tertiary: {
        base: '#45d1f0',
        light: '#00d1ee',
      },
    },
  },

  // Breakpoints
  // Used with `ds.bp()`
  // Keys can be anything you like
  // Have as many breakpoints as you like
  // Values can be use any unit you like
  breakpoints: {
    key: '', // <number | string>,
  },

  // Z-index
  // Used with `ds.z()`
  zIndex: {
    key: 10, // <number>
  },

  // Spacing
  // Used with `ds.spacing()` or `ds.space()`
  spacing: {
    xxxs: '4px',
    xxs: '8px',
    xs: '12px',
    s: '16px',
    m: '24px',
    l: '32px',
    xl: '48px',
    xxl: '64px',
  },
};

export default new DesignSystem(designTokens);
