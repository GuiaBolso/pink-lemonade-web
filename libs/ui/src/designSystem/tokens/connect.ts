import tokens, { TokensGuiaBolsoProps } from './guiaBolso';
import { DeepPartial } from '../../typings/DeepPartial';

export type TokensConnectProps = TokensGuiaBolsoProps &
  DeepPartial<typeof connectTokens>;

const baseFontSize = 16;

const connectTokens = {
  ...tokens,

  name: 'connect',

  type: {
    ...tokens.type,

    baseFontSize,
  },

  colors: {
    ...tokens.colors,

    brand: {
      ...tokens.colors.brand,

      secondary: {
        lightest: '#f2fcfe',
        lighter: '#59d7f2',
        light: '#26caee',
        default: '#00c1eb',
        dark: '#02a6ca',
        darker: '#05829d',
      },
    },
  },
};

export default connectTokens;
