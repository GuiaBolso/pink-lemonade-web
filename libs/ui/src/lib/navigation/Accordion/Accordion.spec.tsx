import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import { Theme } from '../../general';
import { connect, guiaBolso } from '../../../designSystem/tokens';
import { Accordion } from '.';

describe('Text Component', () => {
  const AccordionContent = () => (
    <Theme theme={guiaBolso}>
      <Accordion initOpened summary={<span>Lorem Ipsum</span>}>
        <span data-testid="text-accordion">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          commodo.
        </span>
      </Accordion>
    </Theme>
  );

  const AccordionContentConnect = () => (
    <Theme theme={connect}>
      <Accordion initOpened summary={<span>Lorem Ipsum</span>}>
        <span>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum
          commodo.
        </span>
      </Accordion>
    </Theme>
  );

  it('renders correctly', () => {
    const tree = renderer.create(<AccordionContent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders connect theme correctly', () => {
    const tree = renderer.create(<AccordionContentConnect />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
