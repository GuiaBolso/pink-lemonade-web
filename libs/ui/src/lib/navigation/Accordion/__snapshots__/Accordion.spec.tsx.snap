// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders connect theme correctly 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root testePanel css-123qacc Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-controls="undefined-content"
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiAccordionSummary-root css-16uzaqg Mui-expanded"
    id="undefined-header"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content Mui-expanded"
    >
      <span>
        Lorem Ipsum
      </span>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root css-18o17hk"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="undefined-header"
          id="undefined-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-12ely3a"
          >
            <span>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum commodo.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Text Component renders correctly 1`] = `
<div
  className="MuiPaper-root MuiAccordion-root testePanel css-123qacc Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    aria-controls="undefined-content"
    aria-disabled={false}
    aria-expanded={true}
    className="MuiButtonBase-root MuiAccordionSummary-root css-16uzaqg Mui-expanded"
    id="undefined-header"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiAccordionSummary-content Mui-expanded"
    >
      <span>
        Lorem Ipsum
      </span>
    </div>
    <div
      aria-disabled={false}
      aria-hidden={true}
      className="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role={null}
      tabIndex={null}
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root css-18o17hk"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    className="MuiCollapse-container MuiCollapse-entered"
    style={
      Object {
        "minHeight": "0px",
      }
    }
  >
    <div
      className="MuiCollapse-wrapper"
    >
      <div
        className="MuiCollapse-wrapperInner"
      >
        <div
          aria-labelledby="undefined-header"
          id="undefined-content"
          role="region"
        >
          <div
            className="MuiAccordionDetails-root css-12ely3a"
          >
            <span
              data-testid="text-accordion"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum commodo.
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
